walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,87 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Collection<TimeRange> meetingTimes = new HashSet<>();
+    Collection<TimeRange> meetingTimes = new LinkedHashSet<>();
+    int start = TimeRange.START_OF_DAY;
+    int end = TimeRange.END_OF_DAY;
+    int freeTime;
+    List<TimeRange> unavailableTimes = new ArrayList<>();
+
+    for (Event event : events) {
+        if (!checkRequired(event.getAttendees(), request.getAttendees())) {
+            continue;
+        }
+        unavailableTimes.add(event.getWhen());
+    }
+
+    // Sort unavailable times by their start times.
+    Collections.sort(unavailableTimes);
+
+    for (int i = 0; i < unavailableTimes.size(); i++) {
+      TimeRange time = unavailableTimes.get(i);
+
+      if (i < unavailableTimes.size() - 1) {
+        TimeRange nextTime = unavailableTimes.get(i + 1);
+
+        // Combine current and next times into one time.
+        if (time.overlaps(nextTime)) {
+            int earliest = Math.min(time.start(), nextTime.start());
+            int latest = Math.max(time.end(), nextTime.end());
+
+            time = TimeRange.fromStartEnd(earliest, latest, false);
+            i++;
+        }
+      }
+
+      /** If the event starts in the beginin of the day, make the 
+        * first possible start at the end of this event. 
+        */
```

Comment:
adding this type of comment here is strange. just use multiple lines of `//`

Reply:
Yea i was curious about that